contract_sale_generation/models/contract_line.py

```diff
@@ -40,7 +40,10 @@ def _prepare_sale_line(self, order_id=False, sale_values=False):
             )
             order_line.order_id = sale
         # Get other order line values from product onchange
+        uom = order_line['product_uom']
```

Comment:
Is it possible to send something in context and skip change uom?

```suggestion
        order_line.with_context(skip_change_uom=1).product_id_change()
        
        def product_id_change(self):
            ...
            if self._context.get('skip_change_uom'):
                ...do something...
```

Reply:
No - this would require to change a core function